### ISSUE TITLE
Multiplication not working properly on Helix

### ISSUE THREAD

**Issue by in0ni** [olrtg/emmet-language-server#4]
Hello, I found this here: https://github.com/helix-editor/helix/issues/1896#issuecomment-1636899697, and decided to give it a go (currently using emmet-ls). I'm using helix editor and my config is quite simple (defaults) and works fine. As I was using emmet-ls, I simply changed the `command` and kept the language-server name as it was.

```
[language-server.emmet-ls]
args = ["--stdio"]
command="emmet-language-server"
```

Here is a quick example of what happens (works = shows auto-complete, generates markup):
- div - `works`
- div>heading - `works`
- div>heading* - No autocomplete (good)
- div> heading*2 - Does not work, no auto-complete
- div> heading*2x - `works` *but* don't need the extra `<x>`

I checked the options on your README, perhaps I am missing something?

**Comment by olrtg** [comment]
Hey @in0ni ! How it is going? Hmmm I believe that problem is more related to helix than to this language server. The language server doesn't or shouldn't contain any code editor specific code.

In my neovim setup seem to work the way you want (but again the language server doesn't have any specific code or configuration to make it work with neovim):

![image](https://github.com/olrtg/emmet-language-server/assets/20072509/be438751-4879-4ef9-960e-36e05f8e2079)

Aside from that I don't use helix but maybe you can report or discuss this in the helix repo, maybe is a problem with your config or indeed a problem with helix itself. Still, please let me know if I'm wrong after doing that!

**Comment by in0ni** [comment]
Hey @olrtg, thanks for the quick reply. So you get an auto-complete with this:
`div>heading*2`?

or with `div>heading*2~`, just confused from the screenshot

**Comment by in0ni** [comment]
Hey, so before going to Helix team on this -- with your background on this (I know little about LSP), perhaps these points spark an idea?

- I don't feel it's a config issue, but not ruling it out. A core helix feature is great LSP integration with very simple config in general... all I have defined is the command for the language server, and the arguments as listed above -- all default options.
- switching between `emmet-ls` and `emmet-language-server` only means me changing the value of `command` from above, so shared config
- `emmet-ls` works w/o issues with multiplication -- same emmet syntax, same LSP configuration

So either neovim/helix are sending different lsp requests (of that is what it's called), or emmet-ls/emmet-language-server are doing something different with the request?

**Comment by olrtg** [comment]
I'm also a beginner with language servers so don't worry haha.. oh I see.. that's something I forgot to ask but thanks for confirming that `emmet-ls` works with multiplication without issues.

In theory the completions shouldn't change across editors.. the thing is that this language server is just a wrapper of vscode's emmet implementation so I hope that it is just a problem with the wrapper.. I'll take a look at this later this day.

**Comment by olrtg** [comment]
@in0ni sorry It took longer than expected (busy with work stuff). For me it seems to be working

![image](https://github.com/olrtg/emmet-language-server/assets/20072509/21dc0a74-5e73-4726-9103-ed48aee1b8f5)

I had to trigger completion with Ctrl + x though. Did a simple setup of helix, no multiple language servers (using master build), only emmet-language-server for html was enabled.

Didn't try `emmet-ls` with my setup but It seems to be an issue with completion from helix's part (maybe completion doesn't get triggered by numbers or special chars?). Can you give me some proof that `div>heading*2` is working with `emmet-ls` without triggering completion manually?

Please let me know if that helps!

**Comment by olrtg** [comment]
Also I want to mention that I've released a new version ([v2.0.4](https://github.com/olrtg/emmet-language-server/releases/tag/v2.0.4)) that fixes https://github.com/helix-editor/helix/issues/1896 for this LSP so make sure to update the package to get that fix! 🤘🏽

**Comment by in0ni** [comment]
Hey, I also have been delayed due to work/life. Apparently screen recording is causing my wayland session to crash, which is new and can't look into it now. I played around a little with both `emmet-ls` and `emmet-language-server` and here are some quick points:

### emmet-ls
- always triggers auto-complete, so allows for writing custom markup or xml
- there is no need to press `ctrl-x`, just select the completion (only one option)

### emmet-language-server
- triggers auto-complete for individual elements (valid html)
- requires pressing `ctrl-x` for any emmet syntax that is not a valid html element
- initially, I liked the auto-complete for html elements, but after use found it only gets in the way of writing emmet expressions (would prefer to have the emmet expansion as first item, not last)

I did a few quick examples from this [cheet sheet](https://docs.emmet.io/cheat-sheet/), here is very simple one:
**Syntax**: `div+p`
- emmet-ls: auto-complete comes up automatically, I just press `tab` and I am done.
- emmet-language-server: auto-complete needs to be manually triggered, I have to cycle through multiple options to finally get the emmet expansion (last item) and expand.

Oh, I also updated to 2.0.4 -- nice to see you implement that issue at once!

**Comment by olrtg** [comment]
Hey, thank you so much for taking the time to test this thoroughly! I'll take a look a this in the weekend again to see what I can do.

I'll keep you posted 🤘🏽

**Comment by olrtg** [comment]
Hey! thanks a lot for reporting this 🤘🏽 this should be fixed in https://www.npmjs.com/package/@olrtg/emmet-language-server/v/2.1.0 so please update your package.

If still this doesn't work for you feel free to reopen.

**Comment by youngmoguler** [comment]
After doing according to this repos readme. Every config is fine from my end also done the installation of the emmet-language-server but still it didn't work. Is it because of the path declaration of that npm module which isn't getting activated or what? Because I saw this behavior after installing any npm package globally and done a hx --health [language]  to check and there I saw that package by default wasn't showing for that particular. So, to solve that I only installed lsps that are providing homebrew command.
